components/director/internal/domain/application/service.go

```diff
@@ -301,6 +305,21 @@ func (s *service) Delete(ctx context.Context, id string) error {
 		return errors.Wrapf(err, "while loading tenant from context")
 	}
 
+	scenarios, runtimes, err := s.getScenariosAndRuntimes(ctx, appTenant, id)
+	if err != nil {
+		return errors.Wrapf(err, "while checking if application is in formation and runtime")
+	}
+
+	if len(scenarios) > 0 {
+		msg := fmt.Sprintf("System deletion failed: the system is part of a formation - %s", strings.Join(scenarios, ", "))
+
+		if len(runtimes) > 0 {
+			msg = fmt.Sprintf("%s and of runtime - %s", msg, strings.Join(runtimes, ", "))
+		}
+
+		return apperrors.NewInvalidOperationError(msg)
```

Comment:
i dont think you should be returning an error if the scenario contains no runtimes. With the current impl of scenarios - no runtimes = no one is using the apis of the app

Reply:
This is true, but it is contraversive with what the ticket says:

> Deleting a system which is assigned to a formation must fail

Reply:
UX would be better if when there is no runtime in the scenario, deleting the system should not be stopped.